crates/aptos/src/genesis/keys.rs

```diff
@@ -0,0 +1,101 @@
+        git::{to_base64_encoded_yaml, GitOptions},
+    },
+    op::key,
+    CliCommand,
+};
+use aptos_crypto::{ed25519::Ed25519PrivateKey, x25519, PrivateKey};
+use aptos_types::network_address::DnsName;
+use async_trait::async_trait;
+use clap::Parser;
+use std::path::PathBuf;
+
+const ACCOUNT_KEY_FILE: &str = "account.key";
+const CONSENSUS_KEY_FILE: &str = "consensus.key";
+const NETWORK_KEY_FILE: &str = "network.key";
+
+/// Generate account key, consensus key, and network key for a validator
+#[derive(Parser)]
+pub struct GenerateKeys {
+    #[clap(long, parse(from_os_str))]
+    output_path: PathBuf,
+}
+
+#[async_trait]
+impl CliCommand<Vec<PathBuf>> for GenerateKeys {
+    fn command_name(&self) -> &'static str {
+        "GenerateKeys"
+    }
+
+    async fn execute(self) -> CliTypedResult<Vec<PathBuf>> {
+        let account_key_path = self.output_path.join(ACCOUNT_KEY_FILE);
+        let consensus_key_path = self.output_path.join(CONSENSUS_KEY_FILE);
+        let network_key_path = self.output_path.join(NETWORK_KEY_FILE);
+        let _ = key::GenerateKey::generate_ed25519(EncodingType::Hex, &account_key_path).await?;
+        let _ = key::GenerateKey::generate_ed25519(EncodingType::Hex, &consensus_key_path).await?;
+        let _ = key::GenerateKey::generate_x25519(EncodingType::Hex, &network_key_path).await?;
+        Ok(vec![account_key_path, consensus_key_path, network_key_path])
+    }
+}
+
+/// Upload ValidatorCredentials
+#[derive(Parser)]
+pub struct UploadKeys {
+    /// Github Username for storing names
+    #[clap(long)]
+    github_username: String,
+    #[clap(flatten)]
+    git_options: GitOptions,
+    /// Path to credentials
+    #[clap(long, parse(from_os_str), default_value = ".aptos/")]
+    credentials_path: PathBuf,
+
+    #[clap(long)]
+    validator_host: DnsName,
+    #[clap(long)]
+    validator_port: u16,
+    #[clap(long)]
+    full_node_host: Option<DnsName>,
+    #[clap(long)]
+    full_node_port: Option<u16>,
+}
+
+#[async_trait]
+impl CliCommand<()> for UploadKeys {
+    fn command_name(&self) -> &'static str {
+        "UploadKeys"
+    }
+
+    async fn execute(self) -> CliTypedResult<()> {
+        let account_key_path = self.credentials_path.join(ACCOUNT_KEY_FILE);
+        let consensus_key_path = self.credentials_path.join(CONSENSUS_KEY_FILE);
+        let network_key_path = self.credentials_path.join(NETWORK_KEY_FILE);
+        let account_key: Ed25519PrivateKey =
+            EncodingType::Hex.load_key(ACCOUNT_KEY_FILE, &account_key_path)?;
+        let consensus_key: Ed25519PrivateKey =
+            EncodingType::Hex.load_key(CONSENSUS_KEY_FILE, &consensus_key_path)?;
+        let network_key: x25519::PrivateKey =
+            EncodingType::Hex.load_key(NETWORK_KEY_FILE, &network_key_path)?;
+
+        let user_file_path = format!("{}.yml", self.github_username);
+        let credentials = ValidatorConfiguration {
+            name: self.github_username,
```

Comment:
it seems interesting that this is also the file name, any motivation for the duplication?

Reply:
I wanted it for debugging purposes, but I guess we can cut it